### PR TITLE
[docs] Refactor the debugging tools page

### PR DESCRIPTION
**Pull request by byCedric**
# Why

It was kind of outdated, and didn't include the new inspector method.

# How

This draft focuses on the available tooling starting from SDK 49. It does NOT focus on how to opt-in into this with SDK 48.

- Dropped outdated `Chrome DevTools` section
- Added new `Chrome DevTools` section, with feature hightights
- Updated the `React DevTools` section to focus on the built-in (cli) version
- Added warnings that `Debug remote JS` does not work for Hermes apps

We might need to drop more sections, as they are outdated. I haven't taken a look at the other debugging pages yet, but lets start with this PR.

# Test Plan

Docs change only.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by byCedric** [comment]
Also, I would favor dropping:
1. the `react-native-debugger` section
2. any references to Flipper

File: docs/pages/debugging/tools.mdx [hunk 1]
```diff
@@ -7,17 +7,110 @@ sidebar_title: Tools
...
+
+- iOS device (without USB):
+  - Shake the device.
+  - Touch 3 fingers to the screen.
+- iOS simulator or device (with USB):
+  - Press <kbd>Ctrl</kbd> + <kbd>Cmd ⌘</kbd> + <kbd>z</kbd> or <kbd>Cmd ⌘</kbd> + <kbd>d</kbd>
+
+</Collapsible>
 
 Once you have opened the Developer menu, it will appear as below:
```
Comment (amandeepmittal) [inline docs/pages/debugging/tools.mdx:115]: Unrelated to this PR: the screenshot below this line is outdated.

@byCedric, happy to update this in this or a follow up PR. 

Here is the latest screenshot.

CleanShot 2023-06-12 at 22 43 48@2x

File: docs/pages/debugging/tools.mdx [hunk 2]
```diff
@@ -36,7 +129,9 @@ The Developer menu provides multiple options:
 
 Now let's explore some of the more exciting functionalities.
 
-#### Debug Remote JS
+### Debug Remote JS
```
Comment (amandeepmittal) [inline docs/pages/debugging/tools.mdx:132]: This section doesn't make much sense or relevant (trying to find technical accuracy after this comment [here](https://github.com/expo/expo/pull/22787/files#diff-6edc759c19b235dc978c2960b5dcd0d59782b42d45700b1b46c8ec604f423831R83:~:text=Here%20is%20the%20latest%20screenshot).

Or for apps not using Hermes, it still shows Debug Remote JS?

File: docs/pages/debugging/tools.mdx [hunk 3]
```diff
@@ -7,17 +7,110 @@ sidebar_title: Tools
 import ImageSpotlight from '~/components/plugins/ImageSpotlight';
 import { Terminal } from '~/ui/components/Snippet';
 import Video from '~/components/plugins/Video';
+import { Collapsible } from '~/ui/components/Collapsible';
 
-This page lists a few tools to help debug your Expo project.
+React Native consists of both JavaScript and native code. Making this disctinction is very important when it comes to debugging. If an error is thrown from the JavaScript code, you likely won't be able to find it using debugging tools for native code. This page lists a few tools to help you debug your Expo project.
```
[the remark this answers is not in the document]
Comment (byCedric) [reply docs/pages/debugging/tools.mdx]: Let's move it in a follow-up PR. I think we just want to merge this as is with the last Expo SDK version limitations.

File: docs/pages/debugging/tools.mdx [hunk 4]
```diff
@@ -60,7 +60,7 @@ The Console tab gives you access to an interactive terminal, connected directly
 
 The **Network tab** gives you insights into the network requests made by your app. You can inspect each request and response by clicking on them. This includes `fetch` requests, external loaded media, and in some cases, even requests made by native modules.
 
-This feature is only available when using the dev client and not when using Expo Go.
+> **warning** This feature is only available from Expo SDK 49+ when using the dev client. It is not available when using Expo Go.
```
Comment (amandeepmittal) [inline docs/pages/debugging/tools.mdx:63]:
```suggestion
> **warning** This feature is only available from Expo SDK 49 and higher when using the dev client. It is not available when using Expo Go.
```

Suggestion to use [symbols as words](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#symbols-as-words)

File: docs/pages/debugging/tools.mdx [hunk 5]
```diff
@@ -7,17 +7,110 @@ sidebar_title: Tools
...
+  alt="Enable Pause on exceptions in the right panel of the sources tab"
+  src="/static/images/debugging/inspector-pause-exception.png"
+  style={{ maxWidth: 320 }}
+/>
+
+> **info** Some errors might be caught by other components in your app, like Expo Router. In these cases, you can turn on "Pause on caught exceptions". This will enable you to inspect any thrown error, even when it's handled properly.
+
+### Interacting with the console
+
+The Console tab gives you access to an interactive terminal, connected directly to your app. You can write any JavaScript inside this terminal to execute snippets of code as if it was part of your app. The code is executed in the global scope by default. But, when using breakpoints from the [Source](#pausing-on-breakpoints) tab, it executes in the scope of the reached breakpoint. This allows you to invoke methods and access variables throughout your app.
```
[the remark this answers is not in the document]
Comment (byCedric) [reply docs/pages/debugging/tools.mdx:52]: Yes, but you can set breakpoints "throughout your app" 😄

**Review by amandeepmittal — APPROVED** [review]

**Pull request merged**